driver/src/orderbook/streamed/orderbook.rs

```diff
@@ -0,0 +1,85 @@
+use super::*;
+use crate::{
+    contracts::stablex_contract::batch_exchange,
+    models::{AccountState, Order},
+    orderbook::StableXOrderBookReading,
+};
+use anyhow::Result;
+use ethcontract::{contract::EventData, H256, U256};
+use state::{Batch, State};
+use std::collections::BTreeMap;
+
+// Ethereum events (logs) can be both created and removed. Removals happen if the chain reorganizes
+// and ends up not including block that was previously thought to be part of the chain.
+// However, the orderbook state (`State`) cannot remove events. To support this, we keep an ordered
+// list of all events based on which the state is built.
+
+/// The key by which events are sorted.
+#[derive(Debug, Eq, Ord, PartialEq, PartialOrd)]
+struct Key {
```

Comment:
Not important, but if this was called `EventSortKey` or the like we wouldn't need the comment above. I think generally it's nice to replace comments with more explicit naming (as comments tend to not get updated when the code/logic changes).